pkg/secrethub/auth_aws.go

```diff
@@ -0,0 +1,62 @@
+package secrethub
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/secrethub/secrethub-go/internals/api"
+	"github.com/secrethub/secrethub-go/internals/auth"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sts"
+)
+
+type awsAuthService struct {
+	client client
+}
+
+func newAWSAuthService(client client) AuthMethodService {
+	return &awsAuthService{
+		client: client,
+	}
+}
+
+func (s awsAuthService) Authenticate() (auth.Authenticator, error) {
+	// Currently always use the eu-west-1 region.
+	region := endpoints.EuWest1RegionID
+
+	cfg := aws.NewConfig().WithRegion(region).WithEndpoint("sts." + region + ".amazonaws.com")
+	awsSess, err := session.NewSession(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("could not get AWS session: %v", err)
+	}
+
+	svc := sts.New(awsSess, cfg)
+
+	stsReq, _ := svc.GetCallerIdentityRequest(&sts.GetCallerIdentityInput{})
+
+	err = stsReq.Sign()
+	if err != nil {
+		return nil, fmt.Errorf("could not sign STS request: %v", err)
+	}
+
+	var buf bytes.Buffer
+	err = stsReq.HTTPRequest.Write(&buf)
+	if err != nil {
+		return nil, err
+	}
+
+	req := api.NewAuthRequestAWSSTS(api.SessionTypeHMAC, region, buf.Bytes())
```

Comment:
This raises the question: where is HMAC used? I see now it's the returned authentication type thingy.

Reply:
No clue what this means.

Reply:
Me neither. It feels like I was continuing a conversation I had with myself in another comment, but I cannot find anything that corresponds to this. One thing that catches my eye that I could have meant is: are all `AuthRequestsAWSSTS` of type `HMAC`? In other words, do we want to expose the option to set the `SessionType` with the function or does it default to something?

Tiny thing. Don't spend too much time on it. Resolve it whenever you're satisfied.

Reply:
If we do not want to break this in the future, we have to give it as an option. There currently is only one option.. so it should not be difficult to choose one. Furthermore, I don't think many users will interact with this.

So I am choosing to leave it as is.